tests/test_comparison_template_lib.py

```diff
@@ -508,3 +508,118 @@ def test_postcode_comparison_levels(spark, ctl, Linker):
+                "email": "chris@othermail.com"
+            },
+            { 
+                "unique_id": 4,
+                "email": "chrisa@gmail.com"
+            },
+            { 
+                "unique_id": 5,
+                "email": "chrisa@mali.com"
+            },
+            { 
+                "unique_id": 6,
+                "email": "chrisa@mailtwo.com"
+            },
+            { 
+                "unique_id": 7,
+                "email": "chrisat@verydifferentmail.com"
+            },
+            { 
+                "unique_id": 8,
+                "email": "hcirs@verydifferentmail.com"
+            },
+            { 
+                "unique_id": 9,
+                "email": "christopher@verydifferentmail.com"
+            },
+            { 
+                "unique_id": 10,
+                "email": "notchrisarall@mail.com"
+            },
+            { 
+                "unique_id": 11,
+                "email": "someoneelse@domain.com"
+            },
+            { 
+                "unique_id": 12,
+                "email": "chrismail.com"
+            },
+
+        ]
+    )
+
+    # Generate our various settings objs
+    settings = {
+        "link_type": "dedupe_only",
+        "comparisons": [
+            ctl.email_comparison(
+                col_name="email",
+                levenshtein_thresholds=[2],
+                damerau_levenshtein_thresholds=[2],
+                invalid_emails_as_null=True,
+                include_domain_match_level=True,
+            )
+        ],
+    }
+
+    if Linker == SparkLinker:
+            df = spark.createDataFrame(df)
+            df.persist()
+
+    linker = Linker(df, settings)
+    #Linker = DuckDBLinker(df, settings)
+    linker_output = linker.predict().as_pandas_dataframe()
+
+    print(linker_output)
+
+    # Check individual IDs are assigned to the correct gamma values
+    # Dict key: {gamma_level: tuple of ID pairs}
+    size_gamma_lookup = {
+        9: [(1,2)], #Exact match
+        8: [(1,3), (2,3)], #Exact match on username, different domain
+        7: [(1,4), (2,4)], #Fuzzy match- full email (lev)
+        6: [(1,5), (2,5)], #Fuzzy match- full email (dmlev)
+        5: [(1,6), (2,6)], #Fuzzy match- full email (jw)
+        4: [(1,7), (2,7)], #Fuzzy match- username only (lev)
+        3: [(1,8), (2,8)], #Fuzzy match- username only (dmlev)
+        2: [(1,9), (2,9)], #Fuzzy match- username only (jw)
+        1: [(1,10), (2,10)], #Domain-only match
+        0: [(1,11), (2,11)], #Everything else
+        -1: [(1,12)] #Null level- invalid email
+    }
+
+    for gamma, id_pairs in size_gamma_lookup.items():
+        for left, right in id_pairs:
+            print(f"Checking IDs: {left}, {right}")
+            assert (
+                linker_output.loc[
+                    (linker_output.unique_id_l == left)
+                    & (linker_output.unique_id_r == right)
+                ]["gamma_email"].values[0]
+                == gamma
+            )
```

Comment:
We're using this all over the place now.

Can you migrate it into its own function?

You could even add it to `confest.py` so it can be reused across scripts more easily. Lmk if you need more info on `conftest.py`